### PR TITLE
Updates to swift 4.2. Fixes many linting warnings

### PR DESCRIPTION
**Pull request by rennarda**
I updated to swift 4.2, and fixed the linting issues that were reported from Swift Lint.

### CODE REVIEW

File: Sources/Animator/HeroCoreAnimationViewContext.swift [hunk 1]
```diff
@@ -151,7 +151,7 @@ internal class HeroCoreAnimationViewContext: HeroAnimatorViewContext {
       anim = banim
     }
 
-    anim.fillMode = kCAFillModeBoth
+    anim.fillMode = convertToCAMediaTimingFillMode(convertFromCAMediaTimingFillMode(CAMediaTimingFillMode.both))
```
Comment (nick-potts) [inline Sources/Animator/HeroCoreAnimationViewContext.swift]: Does CAMediaTimingFillMode.both work? seems like you're just converting from a CAMediaTimingFillMode to a string and back.

**Comment by rennarda** [comment]
Sorry - that was cruft added by the automatic conversion. I've gone through and removed the unnecessary code the convertor added, including fully qualified enums where they aren't needed.

**Comment by andreipt** [comment]
Any plans to get this in the next release?

**Comment by SD10** [comment]
Hey @andreipt, 

I don't think there will be a release supporting Swift 4.2 until the Xcode 10 GM release. I think we can try and get this going on a separate branch. I will look into this over the next 2 weeks. I would like to look into this this weekend but I'm a bit behind deadline on a project

**Comment by alexanderkhitev** [comment]
Hi @SD10 any updates?

**Comment by freak4pc** [comment]
Xcode 10 GM is out. Any thoughts?

**Comment by SD10** [comment]
Yes, sorry for the delay everyone. Reviewing this PR now. cc @freak4pc if you can review as well 🙏

**Comment by SD10** [comment]
Closing in favor of #534. Thanks for getting this started @rennarda 👍

**Pull request closed without merging**